homeassistant/components/rflink.py

```diff
@@ -269,11 +272,21 @@ def __init__(self, device_id, hass, name=None, aliases=None, group=True,
         else:
             self._name = device_id
 
+        self._aliases = aliases
+        self._group = group
+        self._group_aliases = group_aliases
+        self._nogroup_aliases = nogroup_aliases
         self._should_fire_event = fire_event
         self._signal_repetitions = signal_repetitions
 
-    def handle_event(self, event):
+    @callback
+    def handle_event_callback(self, entity_id, event):
         """Handle incoming event for device type."""
+        # Make sure event is for this entity
+        # TODO: Can this be avoided, e.g. with device-unique signal?
```

Comment:
Yes, construct the signal using something unique, eg entity_id or device_id. Do this in `async_added_to_hass`. Use the constant already defined as base. Just make it take a string formatting argument.

Reply:
Thanks!